### ISSUE TITLE
Feature Request: Support for COS

### ISSUE THREAD

**Issue by hkantare** [IBM-Cloud/terraform-provider-ibm#629]
Add Terraform support for COS

**Comment by dprosper** [comment]
I believe what you are looking for is here: https://ibm-cloud.github.io/tf-ibm-docs/v0.17.1/d/resource_instance.html

**Comment by hkantare** [comment]
The users are expecting support for COS management via terraform. https://cloud.ibm.com/docs/cloud-object-storage-cli-plugin?topic=cloud-object-storage-cli-ic-cos-cli

**Comment by ThomasUllrich666** [comment]
Any updates here? In which Release Version can we expect COS Support? Is there a roadmap?

**Comment by powellquiring** [comment]
+1 on cos bucket support.  Here is a work around:
```
variable region {
  default = "us-south"
}
variable cos_instance_name {
  default = "vpc-vsi01"
}

locals {
  cos_bucket_region = "${var.region}"
  cos_bucket_name = "${var.cos_instance_name}"
}

resource "ibm_resource_instance" "cos_instance" {
  name              = "${var.cos_instance_name}"
  service           = "cloud-object-storage"
  plan              = "standard"
  location          = "global"
}

resource "null_resource" "bucket" {
  provisioner "local-exec" {
    # Bootstrap script called with private_ip of each node in the clutser
    command = "ibmcloud cos create-bucket --bucket ${local.cos_bucket_name} --ibm-service-instance-id ${ibm_resource_instance.cos_instance.id} --region ${local.cos_bucket_region}"
  }
  provisioner "local-exec" {
    when    = "destroy"
    command = "ibmcloud cos delete-bucket --bucket ${local.cos_bucket_name} --region ${local.cos_bucket_region} --force"
  }
}

output cos_instance_name { value = "${var.cos_instance_name}" }
output cos_instance_id { value = "${ibm_resource_instance.cos_instance.id}" }
output cos_bucket_region { value = "${local.cos_bucket_region}" }
output cos_bucket_name { value = "${local.cos_bucket_name}" }
```

**Comment by mikehollinger** [comment]
+1 for bucket support as well.

I help to lead https://www.ibm.com/us-en/marketplace/ibm-powerai-vision which we're bringing out of private cloud and pushing into public cloud (we're going opposite of a lot of folks) ;-) 

The app creates two directories in the file system today; a ./data which stores the user's images and video, and a ./run which stores our databases. We'd like to put both into COS. 

We'd like to be able to either instantiate (and destroy) a COS bucket if someone uses our trial, OR if they're using our production code, accept a terraform variable that'll let us do something to mount that bucket with s3fs and then decouple the data lifecycle from the VPC instance lifecycle (e.g. someone creates an instance of Vision, does work for 30 min, then destroys it, but their data's persisted in COS).

Our terraform template is pretty simple (it's a single node VM at the moment), with local storage only, if this matters (hence the need for COS-backed storage).

**Comment by hkantare** [comment]
COS Bucket support is available as apart of v0.18.0 release
https://github.com/IBM-Cloud/terraform-provider-ibm/releases/tag/v0.18.0
